### ISSUE TITLE
(aws-cloudfront): Lambda Edge won't delete after 5 days

### ISSUE THREAD

**Issue by jpSimkins** [aws/aws-cdk#15010]
Used the CDK to create a CloudFront distro with a lambda edge function to add security headers. It failed due to another account using the same name (legacy account that has since then been removed). I tried to re-deploy but it wanted me to destroy the stack fist. Well, it's been 5 days and I still cannot destroy the stack. 

I'm aware of the nightmare of Lambda Edge functions and while this may not be a CDK bug, this is an AWS bug to say the least. The Cloudfront distro has been deleted for 5 days. I have no way to manually delete the Lambda Edge functions so I am not not able to deploy my stack. The instructions stated [here](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-edge-delete-replicas.html) do not apply as I have no Distro to edit. The CDK destroy method deleted everything so I am not able to do anything.

Reporting as a bug here because the CDK should have a better implementation of deleting these due to the known limitations in AWS.

I have paid support in other accounts but not this one, so I am not even sure what I can do at this point as not having paid support gives me no way to ask to fix this issue...

### Reproduction Steps

Create a Cloudfront Distro with a Lambda Edge Function. Then delete it.

### What did you expect to happen?

The stack to delete.

### What actually happened?

It deleted everything but the Lambda Edge function and now I am in a state that it will not delete and I have no way to delete it.

```
9:10:18 AM | DELETE_FAILED        | AWS::Lambda::Function         | assetsCdnedgeFunctionFn889F71D0
Resource handler returned message: "Lambda was unable to delete arn:aws:lambda:us-east-1:117781944281:function:CatalogOlyComCdnProd-assetsCdnedgeFunctionFn889F71-0RRGz4brxZvf:1 because it is a replicated function. Please see our documentation for Deleting Lambda@Ed
ge Functions and Replicas. (Service: Lambda, Status Code: 400, Request ID: 8243f0d8-1c06-4175-9872-0743e9041b08, Extended Request ID: null)" (RequestToken: f7c38014-c4ef-f6e3-4d7c-cb97925d08f9, HandlerErrorCode: InvalidRequest)

        new Function (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/@aws-cdk/aws-lambda/lib/function.ts:625:35)
        \_ EdgeFunction.createInRegionFunction (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/@aws-cdk/aws-cloudfront/lib/experimental/edge-function.ts:141:26)
        \_ new EdgeFunction (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/@aws-cdk/aws-cloudfront/lib/experimental/edge-function.ts:60:14)
        \_ CloudFrontWordPressCdn._buildLambdaFunction (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/@olympusat-devops/oly-cloudfront-templates/src/lib/CloudFrontWordPressCdn.ts:116:27)
        \_ new CloudFrontWordPressCdn (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/@olympusat-devops/oly-cloudfront-templates/src/lib/CloudFrontWordPressCdn.ts:64:32)
        \_ new CatalogOlyComCdnProd (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/accounts/oly-marketing-prod/us-east-1/src/stacks/catalog.olympusat.com/src/cdn-prod.ts:11:22)
        \_ Object.<anonymous> (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/accounts/oly-marketing-prod/us-east-1/src/stacks/catalog.olympusat.com/index.ts:29:18)
9:10:18 AM | DELETE_FAILED        | AWS::Lambda::Function         | assetsCdn/edgeFunction/Fn
Resource handler returned message: "Lambda was unable to delete arn:aws:lambda:us-east-1:117781944281:function:CatalogOlyComCdnProd-assetsCdnedgeFunctionFn889F71-0RRGz4brxZvf:1 because it is a replicated function. Please see our documentation for Deleting Lambda@Ed
ge Functions and Replicas. (Service: Lambda, Status Code: 400, Request ID: 8243f0d8-1c06-4175-9872-0743e9041b08, Extended Request ID: null)" (RequestToken: f7c38014-c4ef-f6e3-4d7c-cb97925d08f9, HandlerErrorCode: InvalidRequest)
9:10:19 AM | DELETE_FAILED        | AWS::CloudFormation::Stack    | CatalogOlyComCdnProd
The following resource(s) failed to delete: [assetsCdnedgeFunctionFn889F71D0].
```

### Environment

 - **CDK CLI Version :**  1.102.0 (build a75d52f)
 - **Module Version :**  1.102.0
 - **Node.js Version :**  v14.17.0
 - **OS               :**  Linux Zeus 5.8.0-53-generic #60~20.04.1-Ubuntu SMP Thu May 6 09:52:46 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
 - **Language (Version):** Typescript (6.14.13)
 
--- 

This is :bug: Bug Report

**Comment by jpSimkins** [comment]
6 days now... I cannot re-deploy the stack either
```
CatalogOlyComCdnProd: deploying...
CatalogOlyComCdnProd: creating CloudFormation changeset...

 ❌  CatalogOlyComCdnProd failed: Error [ValidationError]: Stack:arn:aws:cloudformation:us-east-1:117781944281:stack/CatalogOlyComCdnProd/137a63b0-c547-11eb-98d8-0a70beb84243 is in DELETE_FAILED state and can not be updated.
    at Request.extractError (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/aws-cdk/node_modules/aws-sdk/lib/protocol/query.js:50:29)
    at Request.callListeners (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/aws-cdk/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/aws-cdk/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:688:14)
    at Request.transition (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/aws-cdk/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/aws-cdk/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:690:12)
    at Request.callListeners (/home/jpsimkins/DevOps/oly-iac-mono-repo/aws/implementations/node_modules/aws-cdk/node_modules/aws-sdk/lib/sequential_executor.js:116:18) {
  code: 'ValidationError',
  time: 2021-06-08T13:57:20.181Z,
  requestId: '8315373e-ae1e-4d08-89fe-ef782ed1eff6',
  statusCode: 400,
  retryable: false,
  retryDelay: 973.0242701299259
}
Stack:arn:aws:cloudformation:us-east-1:117781944281:stack/CatalogOlyComCdnProd/137a63b0-c547-11eb-98d8-0a70beb84243 is in DELETE_FAILED state and can not be updated.
```

**Comment by njlynch** [comment]
Hi @jpSimkins ,

I've not heard of Lambda@Edge functions persisting for multiple days like this before.

The resolution here likely lands outside the scope of the CDK itself, but here are some things I would try if I were you:
1. `cdk destroy` the stack you want to delete - I see logs above for doing a deploy/update, but attempting a delete/destroy might yield better results.
2. Try to delete the stack via the SDK or Console. In the console, you have the option of deleting the stack and leaving the resources that failed to delete. If you need to re-create the stack, this might be an option.
3. Try posting on the AWS Forums in the [CloudFront forum](https://forums.aws.amazon.com/forum.jspa?forumID=46); if a zombie Lambda@Edge function is still there after a week without any association with a distribution, you might be able to escalate it there.
4. If none of the above work, another workaround might be to rename the function your distribution is using, so it doesn't conflict with the existing zombie function.

Hopefully one of the above solutions will work for you. I don't know that there's anything that the CDK can change here to make the Lambda@Edge deletion process easier, so closing this out for now. If you have follow-ups, feel free to mention me and we can re-open and discuss.

**Comment by jpSimkins** [comment]
Hello @njlynch,

The error I posted in the initial post was the `cdk destroy` response. I added the second error to show that a `cdk deploy` would not work either. I did try to delete the stack manually but I got the same error as was in the initial post. I even tried to delete the lambda function but that wouldn't work either. I did post in the forums as well but no response after a few days either. I was able to contact AWS Support and get them to escalate my issue and get it resolved after 6 days. 

I still wanted to leave this here as this is a common issue I have seen online and given the nature of this issue, the CDK should have a better implementation to prevent these issues. I do plan to _not_ use the Lambda@Edge in the future because of this issue and hope that using CloudFront functions (wasn't available when I wrote the construct) will not have the same affect.

This issue happened due to a failed deployment (name was used in another account) and since that failed deployment, it took 6 days to get this resolved and was only possible with AWS support being involved. Whether this is something the CDK could implement to help with this know issue or if this is an issue with CloudFormation itself. I think this is something that needs to be addressed. If a Lambda@Edge is used, perhaps do not delete the CloudFront distro or at least remove the Lambda@Edge from the distro. 

What I found weird, was even with the distro deleted, the Lambda@Edge failed to delete automatically. Perhaps I was just an edge case but with all I read online, I am not the only one that has had this issue. I do understand this may not be a CDK issue but given the severity of this issue, it needs to be addressed.

Thanks for your time.
